app/routes/chatbot.js

```diff
@@ -112,74 +113,82 @@ router.post('/', (req, res) => {
       });
   });
 
-  const agentViewOip = process.env.MOBILECOMMONS_OIP_AGENTVIEW;
-
-  return loadUser
-    .then((user) => {
-      logger.info(`loaded user:${user._id}`);
-      scope.user = user;
-
-      let campaign;
-      let campaignId;
-
-      if (scope.broadcast_id) {
-        campaignId = process.env.CAMPAIGNBOT_BROADCAST_CAMPAIGN;
-        campaign = app.locals.campaigns[campaignId];
-
-        if (!campaign) {
-          const msg = `Broadcast Campaign '${campaignId}' not found.`;
-          throw new NotFoundError(msg);
+  const loadCampaign = new Promise((resolve, reject) => {
+    logger.log('loadCampaign');
+    let campaign;
+
+    return loadUser
+      .then((user) => {
+        logger.info(`loaded user:${user._id}`);
+        scope.user = user;
+
+        if (scope.broadcast_id) {
+          return app.locals.db.broadcasts
+            .findById(scope.broadcast_id)
+            .then((broadcast) => {
+              campaign = app.locals.campaigns[broadcast.campaign];
+
+              if (!campaign) {
+                const err = new Error('broadcast campaign undefined');
```

Comment:
TODO: Trigger stathat (with email alert) -- broadcast / mData won't work if we don't have a campaign to signup or reportback for.